lib/util/check-token.ts

```diff
@@ -0,0 +1,55 @@
+import { PlatformId } from '../constants';
+import { logger } from '../logger';
+import { GithubReleasesDatasource } from '../modules/datasource/github-releases';
+import { GithubTagsDatasource } from '../modules/datasource/github-tags';
+import type { PackageFile } from '../modules/manager/types';
+import * as hostRules from './host-rules';
+import { parseUrl } from './url';
+
+function isGithubUrl(url: string | null | undefined): boolean {
+  const parsedUrl = parseUrl(url);
+  if (!parsedUrl) {
+    return false;
+  }
+
+  const { hostname } = parsedUrl;
+  return hostname === 'github.com';
```

Comment:
And never api.github.com ?

Reply:
With this function, we'll check only `sourceUrl` and I don't think it can be `api.github.com`. However, will add this anyway.